server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -279,7 +105,7 @@ public static TransportVersion max(TransportVersion version1, TransportVersion v
      * Returns {@code true} if the specified version is compatible with this running version of Elasticsearch.
      */
     public static boolean isCompatible(TransportVersion version) {
-        return version.onOrAfter(MINIMUM_COMPATIBLE);
+        return version.onOrAfter(TransportVersions.MINIMUM_COMPATIBLE);
```

Comment:
Maybe MINIMUM_COMPATIBLE and MINIMUM_CCS_VERSION should continue to be defined here, alongside `current()`, to prevent this cross-referencing?

Reply:
There would still be cross referencing, since the values of MINIMUM_COMPATIBLE and MINIMUM_CCS_VERSION are defined statically. I think current is a bit special, while those constants are fixed (at least for now). If we make them adjustable in the future the way current is, then we could move them back as methods.